dev/tests/api-functional/testsuite/Magento/GraphQl/Swatches/ProductViewTest.php

```diff
@@ -0,0 +1,151 @@
+        $query
+            = <<<QUERY
+{
+  products(filter: {sku: {eq: "configurable"}}) {
+    items {
+      name
+        ... on ConfigurableProduct{
+        configurable_options{
+          id
+          label
+          values {
+            value_index
+            label
+            default_label
+            store_label
+            use_default_value
+          }
+        }
+      }
+    }
+  }
+}
+QUERY;
+        $response = $this->graphQlQuery($query);
+
+        /** @var ProductRepositoryInterface $productRepository */
+        $productRepository = ObjectManager::getInstance()->get(ProductRepositoryInterface::class);
+        $product = $productRepository->get($productSku, false, null, true);
+        /** @var MetadataPool $metadataPool */
+        $metadataPool = ObjectManager::getInstance()->get(MetadataPool::class);
+        $product->setId(
+            $product->getData($metadataPool->getMetadata(ProductInterface::class)->getLinkField())
+        );
+        $this->assertArrayHasKey('products', $response);
+        $this->assertArrayHasKey('items', $response['products']);
+        $configurableOptions = $this->getConfigurableOptions();
+
+        $this->assertBaseFields($response['products'], $configurableOptions);
+    }
+
+    /**
+     * @param $actualResponse
+     * @param $configurableOptions
+     */
+    private function assertBaseFields($actualResponse, $configurableOptions): void
+    {
+        $assertionMap =
+            [
+                'items' => [
+                    [
+                        'name' => 'Configurable Product',
+                        'configurable_options' => [
+                            [
+                                'id' => (int)$configurableOptions['id'],
+                                'label' => $configurableOptions['label'],
+                                'values' => $configurableOptions['options']
+                            ],
+                        ],
+                    ],
+                ],
+            ];
+
+        $this->assertResponseFields($actualResponse, $assertionMap);
+    }
+
+    /**
+     * @return array
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function getConfigurableOptions(): array
+    {
+        if (!empty($this->configurableOptions)) {
+            return $this->configurableOptions;
+        }
+        $productSku = 'configurable';
+        /** @var ProductRepositoryInterface $productRepo */
+        $productRepo = ObjectManager::getInstance()->get(ProductRepositoryInterface::class);
+
+        $product = $productRepo->get($productSku);
+        $configurableAttributeOptions = $product->getExtensionAttributes()->getConfigurableProductOptions();
+        $configurableAttributeOptionsData = [];
+        foreach ($configurableAttributeOptions as $option) {
+            $optionData = $option->getData();
+            $configurableAttributeOptionsData['label'] = $optionData['label'];
+            $cnt = 0;
```

Comment:
Syntaxis of `foreach` operator  is `foreach ($data as $key => $value)`
So we don't need aditional variable `$cnt`